stdlib/builtin/hash.rbs

```diff
@@ -1,410 +1,1033 @@
 
-  def []=: (K arg0, V arg1) -> V
+  public
 
+  # Returns `true` if *hash* is subset of *other*.
+  #
+  #     h1 = {a:1, b:2}
+  #     h2 = {a:1, b:2, c:3}
+  #     h1 < h2    #=> true
+  #     h2 < h1    #=> false
+  #     h1 < h1    #=> false
+  #
   def <: (::Hash[K, V]) -> bool
 
+  # Returns `true` if *hash* is subset of *other* or equals to *other*.
+  #
+  #     h1 = {a:1, b:2}
+  #     h2 = {a:1, b:2, c:3}
+  #     h1 <= h2   #=> true
+  #     h2 <= h1   #=> false
+  #     h1 <= h1   #=> true
+  #
   def <=: (::Hash[K, V]) -> bool
 
+  # Equality---Two hashes are equal if they each contain the same number of keys
+  # and if each key-value pair is equal to (according to Object#==) the
+  # corresponding elements in the other hash.
+  #
+  #     h1 = { "a" => 1, "c" => 2 }
+  #     h2 = { 7 => 35, "c" => 2, "a" => 1 }
+  #     h3 = { "a" => 1, "c" => 2, 7 => 35 }
+  #     h4 = { "a" => 1, "d" => 2, "f" => 35 }
+  #     h1 == h2   #=> false
+  #     h2 == h3   #=> true
+  #     h3 == h4   #=> false
+  #
+  # The orders of each hashes are not compared.
+  #
+  #     h1 = { "a" => 1, "c" => 2 }
+  #     h2 = { "c" => 2, "a" => 1 }
+  #     h1 == h2   #=> true
+  #
+  def ==: (untyped other) -> bool
+
+  # Returns `true` if *other* is subset of *hash*.
+  #
+  #     h1 = {a:1, b:2}
+  #     h2 = {a:1, b:2, c:3}
+  #     h1 > h2    #=> false
+  #     h2 > h1    #=> true
+  #     h1 > h1    #=> false
+  #
   def >: (::Hash[K, V]) -> bool
 
+  # Returns `true` if *other* is subset of *hash* or equals to *hash*.
+  #
+  #     h1 = {a:1, b:2}
+  #     h2 = {a:1, b:2, c:3}
+  #     h1 >= h2   #=> false
+  #     h2 >= h1   #=> true
+  #     h1 >= h1   #=> true
+  #
   def >=: (::Hash[K, V]) -> bool
 
-  def assoc: (K arg0) -> ::Array[K | V]
-
-  # Removes all key-value pairs from *hsh* .
-  # 
-  # ```ruby
-  # h = { "a" => 100, "b" => 200 }   #=> {"a"=>100, "b"=>200}
-  # h.clear                          #=> {}
-  # ```
-  def clear: () -> ::Hash[K, V]
-
-  def compact: () -> ::Hash[K, V]
-
-  def compact!: () -> ::Hash[K, V]?
-
-  # Makes *hsh* compare its keys by their identity, i.e. it will consider
-  # exact same objects as same keys.
-  # 
-  # ```ruby
-  # h1 = { "a" => 100, "b" => 200, :c => "c" }
-  # h1["a"]        #=> 100
-  # h1.compare_by_identity
-  # h1.compare_by_identity? #=> true
-  # h1["a".dup]    #=> nil  # different objects.
-  # h1[:c]         #=> "c"  # same symbols are all same.
-  # ```
-  def compare_by_identity: () -> ::Hash[K, V]
-
-  # Returns `true` if *hsh* will compare its keys by their identity. Also
-  # see `Hash#compare_by_identity` .
-  def compare_by_identity?: () -> bool
+  # Element Reference---Retrieves the *value* object corresponding to the *key*
+  # object. If not found, returns the default value (see Hash::new for details).
+  #
+  #     h = { "a" => 100, "b" => 200 }
+  #     h["a"]   #=> 100
+  #     h["c"]   #=> nil
+  #
+  def []: (K arg0) -> V?
```

Comment:
In `Array`, it returns `V` not `V?`. Writing tests anytime we write `[]` seems overkill.